Explorer/Assets/Scripts/Global/Dynamic/ApplicationParametersParser.cs

```diff
@@ -0,0 +1,147 @@
+        private const string APP_PARAMETER_LOCAL_SCENE = "local-scene";
+        private const string APP_PARAMETER_POSITION = "position";
+
+        public bool LocalSceneDevelopment;
+
+        public readonly Dictionary<string, string> AppParameters = new ();
+
+        public ApplicationParametersParser(RealmLaunchSettings launchSettings)
+        {
+            AppParameters = ParseApplicationParameters();
+
+            if (AppParameters.ContainsKey(APP_PARAMETER_REALM))
+                ProcessRealmAppParameter(launchSettings);
+
+            if (AppParameters.TryGetValue(APP_PARAMETER_POSITION, out string param))
+                ProcessPositionAppParameter(param, launchSettings);
+        }
+
+        private Dictionary<string, string> ParseApplicationParameters()
+        {
+            string[] cmdArgs = Environment.GetCommandLineArgs();
+
+            var deepLinkFound = false;
+            string lastKeyStored = string.Empty;
+
+            for (int i = 0; i < cmdArgs.Length; i++)
+            {
+                string arg = cmdArgs[i];
+
+                if (arg.StartsWith("--"))
+                {
+                    if (arg.Length > 2)
+                    {
+                        lastKeyStored = arg.Substring(2);
+                        AppParameters[lastKeyStored] = string.Empty;
+                    }
+                    else
+                        lastKeyStored = string.Empty;
+                }
+#if !UNITY_EDITOR && UNITY_STANDALONE_WIN
+                else if (!deepLinkFound && arg.StartsWith("decentraland://"))
+                {
+                    deepLinkFound = true;
+                    lastKeyStored = string.Empty;
+
+                    // When started in local scene development mode (AKA preview mode) command line arguments are used
+                    // Example (Windows) -> start decentraland://"realm=http://127.0.0.1:8000&position=100,100&otherparam=blahblah"
+                    ProcessDeepLinkParameters(arg);
+                }
+#endif
+                else if (!string.IsNullOrEmpty(lastKeyStored))
+                    AppParameters[lastKeyStored] = arg;
+            }
+
+            // in MacOS the deep link string doesn't come in the cmd args...
+#if !UNITY_EDITOR && UNITY_STANDALONE_OSX
+            if (!string.IsNullOrEmpty(Application.absoluteURL) && Application.absoluteURL.StartsWith("decentraland"))
+            {
+                // Regex patch for MacOS removing the ':' from the realm parameter protocol
+                ProcessDeepLinkParameters(Regex.Replace(Application.absoluteURL, @"(https?)//(.*?)$", @"$1://$2"));
+            }
+#endif
+
+            return AppParameters;
+        }
+
+        private void ProcessDeepLinkParameters(string deepLinkString)
+        {
+            // Update deep link so that Uri class allows the host name
+            deepLinkString = Regex.Replace(deepLinkString, @"^decentraland:/+", "https://decentraland.com/?");
```

Comment:
isn't it .org?

Reply:
good point Davide, URL should be taken from IDecentralandUrlsSource without hardcoding it

Reply:
Actually we don't need a real url in there, that hardcoded URL being replaced is so that `Uri` class accepts the url as valid to be able to automatically extract all the url params...
It could even be https://www.google.com or whatever that is valid, and it would still accomplish the desired behaviour.

Reply:
But I'll put the `.org` for peace of mind.

Reply:
Updated at: https://github.com/decentraland/unity-explorer/pull/1546/commits/7db392aa7c131317dd50483f83fe60edbcb8d882